packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -78,135 +31,190 @@ export const useGridKeyboardNavigation = (
   const visibleSortedRows = useGridSelector(apiRef, gridVisibleSortedRowEntriesSelector);
   const currentPage = useCurrentPageRows(apiRef, props);
 
-  const mapKey = (event: React.KeyboardEvent) => {
-    if (isEnterKey(event.key)) {
-      return 'ArrowDown';
-    }
-    if (isTabKey(event.key)) {
-      return event.shiftKey ? 'ArrowLeft' : 'ArrowRight';
-    }
-    return event.key;
-  };
+  const goToCell = React.useCallback(
+    (colIndex: number, rowIndex: number) => {
+      logger.debug(`Navigating to cell row ${rowIndex}, col ${colIndex}`);
+      apiRef.current.scrollToIndexes({ colIndex, rowIndex });
+      const field = apiRef.current.getVisibleColumns()[colIndex].field;
+      const node = visibleSortedRows[rowIndex];
+      apiRef.current.setCellFocus(node.id, field);
+    },
+    [apiRef, logger, visibleSortedRows],
+  );
+
+  const goToHeader = React.useCallback(
+    (colIndex: number, event: React.SyntheticEvent<Element>) => {
+      logger.debug(`Navigating to header col ${colIndex}`);
+      apiRef.current.scrollToIndexes({ colIndex });
+      const field = apiRef.current.getVisibleColumns()[colIndex].field;
+      apiRef.current.setColumnHeaderFocus(field, event);
+    },
+    [apiRef, logger],
+  );
 
   const navigateCells = React.useCallback(
     (params: GridCellParams, event: React.KeyboardEvent) => {
       event.preventDefault();
       const dimensions = apiRef.current.getRootDimensions();
-
       if (!currentPage.range || !dimensions) {
         return;
       }
 
-      const colIndex = apiRef.current.getColumnIndex(params.field);
-      const rowIndex = visibleSortedRows.findIndex((row) => row.id === params.id);
-
-      const key = mapKey(event);
-      const isCtrlPressed = event.ctrlKey || event.metaKey || event.shiftKey;
-
-      let nextCellIndexes: GridCellIndexCoordinates;
-      if (isArrowKeys(key)) {
-        nextCellIndexes = getNextCellIndexes(key, {
-          colIndex,
-          rowIndex,
-        });
-      } else if (isHomeOrEndKeys(key)) {
-        const colIdx = key === 'Home' ? 0 : colCount - 1;
-
-        if (!isCtrlPressed) {
-          // we go to the current row, first col, or last col!
-          nextCellIndexes = { colIndex: colIdx, rowIndex };
-        } else {
-          // In that case we go to first row, first col, or last row last col!
-          let newRowIndex = 0;
-          if (colIdx === 0) {
-            newRowIndex = currentPage.range.firstRowIndex;
+      const viewportPageSize = apiRef.current.unstable_getViewportPageSize();
+      const colIndexBefore = params.field ? apiRef.current.getColumnIndex(params.field) : 0;
```

Comment:
Why the ternary operator?

Reply:
Because their is one case where this event is triggered after comiting a row edit and we don't have a field.
It was working because `getColumnIndex` returned `-1` and the clamp behavior set it back  to 0, but it's not great calling it that way.

Reply:
Makes sense. The event below shouldn't be published.

https://github.com/mui-org/material-ui-x/blob/0065e2f89b661070c7536b391544b7e82191698c/packages/grid/_modules_/grid/hooks/features/editRows/useGridEditRows.ts#L460